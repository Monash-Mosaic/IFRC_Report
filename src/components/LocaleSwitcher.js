'use client';

import { usePathname, useRouter } from '@/i18n/navigation';
import { useLocale, useTranslations } from 'next-intl';

export default function LocaleSwitcher() {
  const locale = useLocale();
  const router = useRouter();
  const pathname = usePathname();
  const t = useTranslations('LocaleSwitcher');

  /**
   * Switches the application's locale to the specified new locale.
   *
   * @param {string} newLocale - The locale to switch to.
   */
  const switchLocale = (newLocale) => {
    if (newLocale !== locale) {
      router.replace(pathname, { locale: newLocale });
      router.refresh();
    }
  };
  return (
    <select
      id="locale-switcher"
      name="locale"
      value={locale}
      onChange={(e) => switchLocale(e.target.value)}
      aria-label={t('ariaLabel')}
      title={t('title')}
    >
      <option data-testid="locale-switcher-value-en" value="en">
        🇬🇧 English
      </option>
      <option data-testid="locale-switcher-value-fr" value="fr">
        🇫🇷 Français
      </option>
      <option data-testid="locale-switcher-value-zh" value="zh">
        🇨🇳 中文
      </option>
      <option data-testid="locale-switcher-value-ru" value="ru">
        🇷🇺 Русский
      </option>
      <option data-testid="locale-switcher-value-ar" value="ar">
        🇸🇦 العربية
      </option>
    </select>
  );
}
