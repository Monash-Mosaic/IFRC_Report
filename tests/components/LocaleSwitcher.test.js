import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import LocaleSwitcher from '@/components/LocaleSwitcher';

// Mock navigation helper used by the component
const replaceMock = jest.fn();
const refreshMock = jest.fn();

jest.mock('@/i18n/navigation', () => ({
  useRouter: () => ({ replace: replaceMock, refresh: refreshMock }),
  usePathname: () => '/current-path',
}));

// Mock next-intl hooks used by the component
let currentLocale = 'en';
jest.mock('next-intl', () => ({
  useLocale: () => currentLocale,
}));

describe('LocaleSwitcher', () => {
  beforeEach(() => {
    replaceMock.mockClear();
    refreshMock.mockClear();
    currentLocale = 'en';
  });

  it(`renders LocaleSwitcher`, async () => {
    const { container } = render(<LocaleSwitcher />);
    expect(container).toMatchSnapshot();
  });
});
