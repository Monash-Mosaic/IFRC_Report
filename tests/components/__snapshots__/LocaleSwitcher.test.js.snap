// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`LocaleSwitcher renders LocaleSwitcher 1`] = `
<div>
  <select>
    <option
      data-testid="locale-switcher-value-en"
      value="en"
    >
      🇬🇧 English
    </option>
    <option
      data-testid="locale-switcher-value-fr"
      value="fr"
    >
      🇫🇷 Français
    </option>
    <option
      data-testid="locale-switcher-value-zh"
      value="zh"
    >
      🇨🇳 中文
    </option>
    <option
      data-testid="locale-switcher-value-ru"
      value="ru"
    >
      🇷🇺 Русский
    </option>
    <option
      data-testid="locale-switcher-value-ar"
      value="ar"
    >
      🇸🇦 العربية
    </option>
  </select>
</div>
`;
